src/math/bandmat.rs

```diff
@@ -62,6 +63,18 @@ impl<T> BandMat<T> {
     pub unsafe fn push(&mut self, val: T) {
         self.data.push(val);
     }
+
+    pub unsafe fn from_matrix(mut mat: Mat<T>, sub_diagonals: c_int, sup_diagonals: c_int) -> BandMat<T> {
+        let data = mat.as_mut_ptr();
+        let length = mat.cols() * mat.rows();
```

Comment:
How are overflow conditions checked in other code sections?

Reply:
The other sections (see for example `src/math/mat.rs`) don't actually do any overflow checking. Perhaps they should be added on a higher level, since in this section we are only copying one array into another. Perhaps in the constructor we can check that `rows * cols` does not overflow?

Reply:
That would make most sense, are there any other `safe` methods modifying either row or col count? If so those would need to be checked too.